ale_linters/go/gometalinter.vim

```diff
@@ -10,11 +10,11 @@ endfunction
 
 function! ale_linters#go#gometalinter#GetCommand(buffer) abort
     let l:executable = ale_linters#go#gometalinter#GetExecutable(a:buffer)
-    let l:filename = expand('#' . a:buffer . ':p')
+    let l:filename = expand('#' . a:buffer)
     let l:options = ale#Var(a:buffer, 'go_gometalinter_options')
 
     return ale#Escape(l:executable)
-    \   . ' --include=''^' . l:filename . '.*$'''
+    \   . ' --include=''' . l:filename . ''''
```

Comment:
Could you change this to `' --include=' . ale#Escape(l:filename)`? If the filename contains single quotes, it won't be escaped properly here.

`--include` takes a regular expression, doesn't it? We should escape the characters in the filename. `\` and `.` will appear pretty commonly, and have special meaning in regex.